homeassistant/components/fritz/common.py

```diff
@@ -311,6 +311,16 @@ def unregister_entity_updates() -> None:
                 )
         return unregister_entity_updates
 
+    def entity_states_update(self) -> dict:
+        """Run registered entity update calls."""
+        entity_states = {}
+        for key in list(self._entity_update_functions):
+            _LOGGER.debug("update entity %s", key)
+            entity_states[key] = self._entity_update_functions[key](
```

Comment:
It's weird that we copy the keys to a list above but don't have a guard here when we access the key in the dict. If we're worried about the dict changing while we're iterating it we'd need to have some guard here too.

Reply:
we had some issues in the past (_especial during startup on slower systems_) that while the coordinator update is running, new entities has been registered, those the `self._entity_update_functions` get changed - so there might be some entries added, but never removed from this dict

Reply:
When an entity is removed the update function may be removed, right?

Reply:
there is no automatic entity deletion mechanic at the moment in the `fritz` integration - only a reload or stop of the integration could cause this

Reply:
I'd not rely on the entity not getting removed under some circumstances. The entity registry can remove the entity and this may be exposed to the user in more places in the future.

Reply:
ah ok - was not aware of that the user in future will be able to remove entities which are still provided by the integration.
i've added a guard